File: frontend/lib/domain/entities/service_category.dart
import 'package:equatable/equatable.dart';

class ServiceCategoryEntity extends Equatable {
  const ServiceCategoryEntity({
    required this.id,
    required this.name,
    required this.showInNavbar,
    required this.isActive,
    this.description,
    this.displayOrder = 0,
    this.createdAt,
    this.updatedAt,
  });

  final String id;
  final String name;
  final String? description;
  final bool showInNavbar;
  final int displayOrder;
  final bool isActive;
  final DateTime? createdAt;
  final DateTime? updatedAt;

  @override
  List<Object?> get props => [
        id,
        name,
        description,
        showInNavbar,
        displayOrder,
        isActive,
        createdAt,
        updatedAt,
      ];
}

File: backend_python/app/models/service_category.py
"""
Service Category model
"""
from datetime import datetime
from typing import Optional
from beanie import Document
from pydantic import Field


class ServiceCategory(Document):
    """Service Category model"""
    name: str = Field(..., unique=True)
    description: Optional[str] = None
    show_in_navbar: bool = Field(default=False, alias="showInNavbar")
    display_order: int = Field(default=0, alias="displayOrder")
    is_active: bool = Field(default=True, alias="isActive")
    created_at: datetime = Field(default_factory=datetime.utcnow, alias="createdAt")
    updated_at: datetime = Field(default_factory=datetime.utcnow, alias="updatedAt")
    
    class Settings:
        name = "ServiceCategory"
        # Note: Indexes are managed by Prisma, so we don't define them here to avoid conflicts
        # indexes = [("showInNavbar", "isActive"), "displayOrder"]

